src/main/java/uk/gov/pay/directdebit/mandate/services/MandateStateUpdateService.java

```diff
@@ -50,6 +51,7 @@ void mandateCreatedFor(Mandate mandate) {
     Mandate confirmedOnDemandDirectDebitDetailsFor(Mandate mandate) {
         Mandate updatedMandate = confirmedDetailsFor(mandate);
         userNotificationService.sendOnDemandMandateCreatedEmailFor(updatedMandate);
+        directDebitEventService.registerDirectDebitConfirmedEventFor(updatedMandate);
```

Comment:
Why did this have to be added?

Reply:
The only place this method to register a DDEvent when confirming a mandate was in the test further down (which didn't work as expected and therefor never alerted us to the fact that this method was never being called when confirming a mandate). Therefor I believe we should be registering a confirm mandate event when we confirm a mandate, so I've added it here.